File: Samples/GridLoaders/ASP.NET/VB/GridLoader/Navigation.ascx.vb
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Collections
Imports System.IO
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls

Namespace EntitySpaces.Websites.GridLoader
    Partial Public Class Navigation
        Inherits System.Web.UI.UserControl
        Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            CreateIndex()
        End Sub

        Private Sub CreateIndex()
            Try
                Dim di As New DirectoryInfo(Server.MapPath(Constants.GeneratedControlsDirectory))
                Dim files As FileInfo() = di.GetFiles(String.Concat("*", Constants.ControlFileExtenstion), SearchOption.TopDirectoryOnly)

                For Each file As FileInfo In files
                    Dim link As New HyperLink()
                    link.NavigateUrl = Utilities.ConstructUrl(Constants.ControlKey, file.Name.Replace(String.Concat(Constants.ControlPostfix, Constants.ControlFileExtenstion), String.Empty))

                    link.Text = file.Name.Replace(String.Concat(Constants.ControlPostfix, Constants.ControlFileExtenstion), String.Empty)

                    NavigationMenuStripPlaceHolder.Controls.Add(link)
                    NavigationMenuStripPlaceHolder.Controls.Add(New LiteralControl("&nbsp;&nbsp;&nbsp;"))
                Next
            Catch exc As Exception
                'Exceptions.ProcessModuleLoadException(this, exc);

            End Try
        End Sub
    End Class
End Namespace

File: Samples/GridLoaders/ASP.NET/C#/GridLoader/Navigation.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.IO;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

namespace EntitySpaces.Websites.GridLoader
{
    public partial class Navigation : System.Web.UI.UserControl
    {
        protected void Page_Load(object sender, EventArgs e)
        {
            CreateIndex();
        }

        private void CreateIndex()
        {
            try
            {
                DirectoryInfo di = new DirectoryInfo(Server.MapPath(Constants.GeneratedControlsDirectory));
                FileInfo[] files = di.GetFiles(string.Concat("*", Constants.ControlFileExtenstion), SearchOption.TopDirectoryOnly);

                foreach (FileInfo file in files)
                {
                    HyperLink link = new HyperLink();
                    link.NavigateUrl = Utilities.ConstructUrl(Constants.ControlKey,
                      file.Name.Replace(string.Concat(Constants.ControlPostfix, Constants.ControlFileExtenstion), string.Empty));

                    link.Text = file.Name.Replace(
                        string.Concat(Constants.ControlPostfix, Constants.ControlFileExtenstion), string.Empty);

                    NavigationMenuStripPlaceHolder.Controls.Add(link);
                    NavigationMenuStripPlaceHolder.Controls.Add(new LiteralControl("&nbsp;&nbsp;&nbsp;"));
                }
            }

            catch (Exception exc)
            {
                //Exceptions.ProcessModuleLoadException(this, exc);
            }
        }
    }
}